secio/src/stream_cipher.rs

```diff
@@ -39,18 +43,40 @@ impl Cipher {
 /// Returns your stream cipher depending on `Cipher`.
 #[inline]
 pub fn ctr_init(key_size: Cipher, key: &[u8], iv: &[u8]) -> StreamCipher {
-    match key_size {
-        Cipher::Aes128 => Box::new(Aes128Ctr::new(
-            GenericArray::from_slice(key),
-            GenericArray::from_slice(iv),
-        )),
-        Cipher::Aes256 => Box::new(Aes256Ctr::new(
-            GenericArray::from_slice(key),
-            GenericArray::from_slice(iv),
-        )),
-        Cipher::TwofishCtr => Box::new(Ctr128::<Twofish>::new(
-            GenericArray::from_slice(key),
-            GenericArray::from_slice(iv),
-        )),
+    INIT.call_once(|| unsafe {
+        AES_NI = is_x86_feature_detected!("aes") && is_x86_feature_detected!("sse3");
+    });
+
+    let enable = unsafe { AES_NI };
+    if enable {
```

Comment:
Is the `enable` needed, what's the problem of using `AES_NI` directly?

Reply:
Just need an unsafe wrapper around AES_NI to get the value because AES_NI is mut.

I am thinking about whether to use a Cell<bool> or AtomicBool , so that the static variable does not need to be labeled mut

Reply:
I see. we could use `lazy_static` instead of `INIT.call_once`, in this way the `mut` could be removed.

Reply:
`lazy_static` is `Cell<T>` and `Once`.